src/libmongoc/tests/test-mongoc-sample-commands.c

```diff
@@ -2474,6 +2474,182 @@ test_example_56 (mongoc_database_t *db)
+      goto cleanup;
+   }
+
+cleanup:
+   bson_destroy (doc);
+   return rc;
+}
+
+
+static bool
+pet_setup (mongoc_collection_t *cats_collection,
+           mongoc_collection_t *dogs_collection)
+{
+   bool ok = true;
+
+   mongoc_collection_drop (cats_collection, NULL);
+   mongoc_collection_drop (dogs_collection, NULL);
+
+   ok = insert_pet (cats_collection, true);
+   if (!ok) {
+      goto done;
+   }
+
+   ok = insert_pet (dogs_collection, true);
+   if (!ok) {
+      goto done;
+   }
+
+   ok = insert_pet (dogs_collection, false);
+   if (!ok) {
+      goto done;
+   }
+done:
+   return ok;
+}
+
+
+/*
+ * Increment 'accumulator' by the amount of adoptable pets in the given
+ * collection.
+ */
+static bool
+accumulate_adoptable_count (mongoc_collection_t *collection,
+                            long long *accumulator /* OUT */
+)
+{
+   bson_t *pipeline = NULL;
+   mongoc_cursor_t *cursor = NULL;
+   bool rc;
+   const bson_t *doc = NULL;
+   bson_error_t error;
+   const bson_value_t *value = NULL;
+   bson_iter_t iter;
+
+   pipeline = BCON_NEW ("pipeline",
+                        "[",
+                        "{",
+                        "$match",
+                        "{",
+                        BCON_UTF8 ("adoptable"),
+                        BCON_BOOL ("true"),
+                        "}",
+                        "}",
+                        "{",
+                        "$count",
+                        BCON_UTF8 ("adoptableCount"),
+                        "}",
+                        "]");
+
+   cursor = mongoc_collection_aggregate (
+      collection, MONGOC_QUERY_NONE, pipeline, NULL, NULL);
+
+   rc = mongoc_cursor_next (cursor, &doc);
+
+   if (mongoc_cursor_error (cursor, &error)) {
+      MONGOC_ERROR ("could not get adoptableCount: %s", error.message);
+      rc = false;
+      goto cleanup;
+   }
+
+   if (!rc) {
+      MONGOC_ERROR ("%s", "cursor has no results");
+      goto cleanup;
+   }
+
+   rc = bson_iter_init_find (&iter, doc, "adoptableCount");
+   if (rc) {
+      value = bson_iter_value (&iter);
+      if (value->value_type == BSON_TYPE_INT32) {
+         *accumulator += value->value.v_int32;
+      } else if (value->value_type == BSON_TYPE_INT32) {
+         *accumulator += value->value.v_int64;
+      } else {
+         MONGOC_ERROR ("%s", "'adoptableCount' must be an integer");
+         rc = false;
+         goto cleanup;
+      }
+   } else {
+      MONGOC_ERROR ("%s", "missing key: 'adoptableCount'");
+      goto cleanup;
+   }
+
+cleanup:
+   bson_destroy (pipeline);
+   mongoc_cursor_destroy (cursor);
+   return rc;
+}
+
+/*
+ * JIRA: https://jira.mongodb.org/browse/DRIVERS-2181
+ */
+static void
+test_example_59 (mongoc_database_t *db)
+{
+   /* Start Example 59 */
+   mongoc_client_t *client = NULL;
+   mongoc_client_session_t *cs = NULL;
+   mongoc_collection_t *cats_collection = NULL;
+   mongoc_collection_t *dogs_collection = NULL;
+   long long adoptable_pets_count = 0;
```

Comment:
One note on this, some platforms define int64_t to be `long`, while others define it to be `long long`, so portable print specifiers get a little hairy. There is a portable macro `PRId64` that solves this, but the code is just a little ugly.

```C
   printf ("there are %" PRId64 " adoptable pets\n", adoptable_pets_count);
```
This macro would fix the following warning:
```
[2022/11/14 19:18:51.748] /data/mci/53a76e388c58b50897191ae23d61ac25/mongoc/src/libmongoc/tests/test-mongoc-sample-commands.c:2639:26: error: format '%lld' expects argument of type 'long long int', but argument 2 has type 'int64_t {aka long int}' [-Werror=format=]
[2022/11/14 19:18:51.748]     printf ("there are %lld adoptable pets\n", adoptable_pets_count);
[2022/11/14 19:18:51.748]                        ~~~^
[2022/11/14 19:18:51.748]                        %ld
```

Leaving the variable as a `long long` would also be portable since a `long long` is defined to be at least 64 bits. Which way would you prefer?

Reply:
We use `int64_t` in the codebase. I think it is safe to assume users have stdint.h. I prefer `int64_t` since it matches `bson_iter_as_int64` return.